Porting g3log to qnx

rtimmaraju opened the pull request:
qnx source changes.

Commit:
g3log : ported and UT tested for  g3log on  QNX envionment

Commit:
g3log: 7.1 compiled version

Commit:
Removing the crashhandler from the QNX build and removed the tests having those dependencies

Commit:
g3log:removing extra spaces and blank lines and adding qnx flag to the test_fatal testcases

Commit:
g3log:refactoring the g3log changes

Commit:
g3log:performance.cmake kept unchanged

Commit:
disabled the stackdump from qnx

eleir9268 commented on Build.cmake:
```diff
@@ -17,11 +17,18 @@ file(GLOB HEADER_FILES ${LOG_SRC}/g3log/*.hpp)
+        ${LOG_SRC}/g3log/stacktrace_windows.hpp
+    )
+endif()
+
```
We usually don't touch things unless it's necessary, and those we, usually, wrap in a if (QNX) block.

If you tried to get these changes approved by upstream there's a good chance they would tell you to go back and fix this.

eleir9268 commented on src/crashhandler_unix.cpp:
```diff
@@ -77,7 +79,13 @@ namespace {
          fatal_stream << "(" << signal_number << ")\tPID: " << getpid() << std::endl;
          fatal_stream << "\n***** SIGNAL " << fatal_reason << "(signo= " << signal_number << " si_errno= " << info->si_errno << " si_code = " << info->si_code << ")" << std::endl;
+		 
+		 #if (defined(__QNX__))
```
These preprocessor directives should not be indented.

eleir9268 commented on test_unit/Test.cmake:
```diff
@@ -23,46 +23,27 @@
-                  "${gmock_SOURCE_DIR}/include")
-      endif()
-
-      enable_testing()
```
Wrap this in a `if (NOT QNX)` block.

rtimmaraju replied:
tests build for three environments MSVC , MINGW and LiNUX   , so added  specific to qnx if ( QNX) , rest all goes to else.

eleir9268 replied:
Looks good, but the indentation should be fixed now.

eleir9268 commented on test_unit/Test.cmake:
```diff
@@ -76,24 +57,38 @@
-		add_test( ${test} ${test} )
+      ENDIF()
+
+		  add_test( ${test} ${test} )
```
The indentation here looks wrong.

eleir9268 commented on test_unit/Test.cmake:
```diff
@@ -76,24 +57,38 @@
        set_target_properties(test_dynamic_loaded_shared_lib PROPERTIES COMPILE_DEFINITIONS "GTEST_HAS_RTTI=0")
-       target_link_libraries(test_dynamic_loaded_shared_lib  ${G3LOG_LIBRARY} -ldl gtest_main)
+       target_link_libraries(test_dynamic_loaded_shared_lib  ${G3LOG_LIBRARY} -ldl gtest_main) 
+
```
Unnecessary whitespace.

eleir9268 commented on test_unit/test_fatal.cpp:
```diff
@@ -64,7 +64,7 @@ TEST(LogTest, FatalSIGTERM__UsingDefaultHandler) {
 }
 
-#if !(defined(WIN32) || defined(_WIN32) || defined(__WIN32__))
+#if !(defined(WIN32) || defined(_WIN32) || defined(__WIN32__) )
```
Unnecessary whitespace.

Commit:
g3log:Removel of unwanted changes

nnarayanamurthy commented on test_unit/Test.cmake:
```diff
@@ -23,6 +23,10 @@
    # =========================
    IF (ADD_G3LOG_UNIT_TEST)
+   	if(QNX)
+    		set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
```
please check the indentation

nnarayanamurthy commented on test_unit/Test.cmake:
```diff
@@ -48,21 +52,25 @@
 
-      SET(tests_to_run test_message test_filechange test_io test_fatal test_signal test_cpp_future_concepts test_concept_sink test_sink ${OS_SPECIFIC_TEST})
+       
+     SET(tests_to_run test_message test_filechange test_io test_fatal test_signal test_cpp_future_concepts test_concept_sink test_sink ${OS_SPECIFIC_TEST})
```
i think there is no change just some extra lines and indentation added, please check once

nnarayanamurthy commented:
Please review all the changes and fix any indentation, extra whitespace, or unnecessary blank lines.

Commit:
g3log:review comments are sdressed

Commit:
g3log:review comments are adressed2

felixlionardo-qnx commented on src/crashhandler_unix.cpp:
```diff
@@ -77,8 +79,14 @@ namespace {
-         trigger.stream() << fatal_stream.str();
+#endif
+
+      trigger.stream() << fatal_stream.str();
```
Unnecessary white space after #endif and wrong indentation

Commit:
indentation corrected

dchordia committed:
Fix multiple issues
 - Fix intendation, formatting, styling, install locations
 - Add test install option
 - Remove unnecessary changes

dchordia committed:
Add back newline

dchordia committed:
Add install subdir for clarity

dchordia reviewed (approved):
LGTM
